libsolidity/interface/StandardCompiler.cpp

```diff
@@ -208,6 +209,23 @@ bool isBinaryRequested(Json::Value const& _outputSelection)
 	return false;
 }
 
+/// @returns true if any Yul IR was requested. Note that as an exceptin, '*' does not
+/// yet match "ir" or "irOptimized"
+bool isIRRequested(Json::Value const& _outputSelection)
```

Comment:
The beauty of this function is not even topped by the most sophisticated async javascript code I have seen so far ;)